VMAccess/vmaccess.py

```diff
@@ -397,29 +399,26 @@ def _fsck_check(hutil):
         hutil.do_exit(1, 'Check', 'error', '0', 'Check failed.')
 
 
-def _fsck_repair(hutil):
+def _fsck_repair(disk_name, hutil):
     # first unmount disks and loop devices lazy + forced
     try:
-        cmd_result = waagent.Run("umount -adfr")
+        cmd_result = waagent.Run("umount -f /%s" % disk_name)
         if cmd_result!=0:
             # Fail fast
-            raise Exception("Failed to unmount disks")
+            hutil.log("Failed to unmount disk")
+            raise Exception("Failed to unmount disk %s" % disk_name)
```

Comment:
maybe soft fail here? since the disk may not have mounted at all if it was 'sufficiently' broken...
Then the 'else' on the next line should not be there and you should always fsck.